lib/vtls/darwinssl.c

```diff
@@ -876,6 +876,39 @@ CF_INLINE CFStringRef CopyCertSubject(SecCertificateRef cert)
   return server_cert_summary;
 }
 
+static CURLcode CopyCertSubject(struct Curl_easy *data,
+                                SecCertificateRef cert, char **certp)
+{
+  CFStringRef c = getsubject(cert);
+  CURLcode result = CURLE_OK;
+  char *cbuf = NULL;
+  *certp = NULL;
+
+  /* If subject is not UTF-8 then check if it can be converted */
+  if(!CFStringGetCStringPtr(c, kCFStringEncodingUTF8)) {
```

Comment:
Sorry I just saw this. It looks like you've changed the behavior so certp receives the subject as a c string in UTF-8 encoding but currently that won't happen if subject is already UTF-8. Can you guys please review #1836 for more details and a potential fix.

Another thing but not critical is certp could be optional so that if this function is only called to check if encoding UTF-8 succeeds then there could be no unnecessary copy.